### DISCUSSION TITLE
How do I add constraints between list values?

### DISCUSSION THREAD

**Discussion by emil79** [cuelang/cue#942] [Q&A]
I would like to constrain some list elements. I have some geographic bounding boxes that I'd like to validate are stored as `[[south, west], [north, east]]`

I have a definition like this:
```
#LatLng: [number, number]
#BoundingBox: [#LatLng, #LatLng]
```

What I want is to add is something like
```
#BoundingBox: [[south, west], [>south, >west]]
```
but it doesn't work. Is anything like this possible?

**Comment by mpvl** [comment]
We have considered adding list element aliases before. Now with the addition of value aliases, it would make sense to add these.

This would allow writing:

```
#BoundingBox: [X=[south, west], [>X.south, >X.west]]
```

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/942.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
